### ISSUE TITLE
Entity with OwnsOne property providing not optimal SQL

### ISSUE THREAD

**Issue by iuridosanjos** [dotnet/efcore#20763]
### Database entities

```
public class Customer
{
	public int Id { get; set; }
	public string Name { get; set; }
	public Address Address { get; set; }
}

public class Address
{
	public string StreetAddress { get; set; }
	public string City { get; set; }
	public string State { get; set; }
	public string ZipCode { get; set; }
}
```

### DbContext

```
public class MyContext : DbContext
{
	public DbSet<Customer> Customers { get; set; }

	protected override void OnModelCreating(ModelBuilder modelBuilder)
	{
		base.OnModelCreating(modelBuilder);

		modelBuilder.Entity<Customer>().OwnsOne(c => c.Address);
	}
}
```

### Problem
When I load all the Costumers from the Database, it runs the following SQL:

```
SELECT [c].[Id], [c].[Name], [t].[Id], [t].[Address_City], [t].[Address_State], [t].[Address_StreetAddress], [t].[Address_ZipCode]
FROM [Customers] AS [c]
LEFT JOIN (
    SELECT [c0].[Id], [c0].[Address_City], [c0].[Address_State], [c0].[Address_StreetAddress], [c0].[Address_ZipCode]
    FROM [Customers] AS [c0]
    WHERE [c0].[Address_ZipCode] IS NOT NULL OR ([c0].[Address_StreetAddress] IS NOT NULL OR ([c0].[Address_State] IS NOT NULL OR [c0].[Address_City] IS NOT NULL))
) AS [t] ON [c].[Id] = [t].[Id]
```

Note the unneeded and undesired nested select just to get _Address_ properties. I want EF Core to load all of _Address_ properties as if _Address_ could not be null (the internal properties could be null, but not Customer.Addresss), but _OwnedNavigationBuilder<TEntity, TRelatedEntity>_ don't have anything like "_IsRequired_" (for the _Customer.Address property_). How can I do that?

The SQL should look like this:

```
SELECT [c].[Id], [c].[Name], [c].[Address_City], [c].[Address_State], [c].[Address_StreetAddress], [c].[Address_ZipCode]
FROM [Customers] AS [c]
```

That way, I wanted to always create and "_new Address()_" with those results.

Can I change anything in my model configuration to tell EF Core to behave like this?

### Note
This case is just an example, but shows exactly the problem I'm having in my product. The basic difference is that I'm using the Owned class for multiple properties, same as if the shown example used like 5 to 10 properties of that type (not collection). The SQL that EF Core is executing is not good and it's performancing poorly than we expected it should.

### Technical Information

EF Core Version tested here: 3.1.3
SQL Server Version: SQL Server 2019
Using .NET Core 3.1

**Comment by ajcvickers** [comment]
@iuridosanjos This looks like a duplicate of #18299, which has been fixed in [EF Core 5.0 preview 3](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/5.0.0-preview.3.20181.2). Can you try preview 3 and report back if it does not fix the issues you are seeing?

**Comment by iuridosanjos** [comment]
@ajcvickers I've tested on the EF Core 5.0 preview 3 you've said and it's fixed. The SQL is not making unnecessary JOINs, the result of the above example is only one simple SELECT.

Thanks.

**Comment by dr130873** [comment]
Any plans of putting this fix on EF core 3.1

**Comment by ajcvickers** [comment]
@dr130873 As discussed on #18299, it is not feasible to port this change to 3.1.

**Comment by MikeStonoga** [comment]
> @iuridosanjos This looks like a duplicate of #18299, which has been fixed in [EF Core 5.0 preview 3](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/5.0.0-preview.3.20181.2). Can you try preview 3 and report back if it does not fix the issues you are seeing?

That works for me!
Thanks a lot!
A simple query downgrade from 4s to 0.25s
